### ISSUE TITLE
Fail if audience is specified but absent in the token.

### ISSUE THREAD

**Issue by mhils** [auth0/java-jwt#33]
Hi there,

reading through the source, why does the following not throw an exception?
```java
new JWTVerifier(secret, "audience").verify("my-token-without-aud");
```

If I explicitly require an audience in the verifier, this check should not be skipped if there's no audience specified in the token? Similarly, the same behaviour applies to the issuer attribute.

**Comment by ojacobson** [comment]
This applies to issuer, as well.

**Comment by hzalaz** [comment]
Fixed in #57
